UofS-Pulse-Binfo/analyzedphenotypes
# Issue #4   distribution chart enhancement and select fields not working

Fixed issues in data visualizer.

To test:
1. Select Dataset Form.
This form supports either field selected to create a distribution chart. When experiment is selected the trait field will sort option and push all relevant trait up the top of the list of options (emphasized by dark coloured text) while non-related traits will be at the bottom list with lighter colour. However, non-related traits are still selectable in the event user decides to change. On the other hand, when user selects a trait, experiment field updates to the corresponding experiment instantly. 

2. When experiment and trait both have values, click Generate Chart button.
In this stage user can select other traits in the project. In the experiment field, the option
"Select Experiment" will take user to Select Dataset form to restart the process or select a different project.

3. While in 2, the url contains project id and trait id.
I have added a check so that if this values are altered to either a string (we are expecting integer) or an id number that does not exist, form Select Dataset will load and let user select a project/trait.
